### PR TITLE
feat: Add more status for machine workflow

### PR DESCRIPTION
**Pull request by helayoty**
-  Add more status for machine creation, check status, and install plugins.
- Move `ensureNodePlugins` to the workspace controller to be able to update the status when waiting for plugins installed.

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -53,20 +55,36 @@ func ListNodes(ctx context.Context, kubeClient client.Client, options *client.Li
...
-		return err
-	}
-	//does node have the custom label for DADI
-	foundDADIPlugin, err := checkAndInstallDADI(ctx, nodeObj, kubeClient)
-	if err != nil {
-		return err
-	}
-	if foundNvidiaPlugin && foundDADIPlugin {
-		return nil
+
```
Comment (Fei-Guo) [inline pkg/node/node.go]: I have one suggestion to move all add label (for nvidia and dadi) code here before the wait.

File: pkg/node/node.go [hunk 2]
```diff
@@ -53,20 +55,36 @@ func ListNodes(ctx context.Context, kubeClient client.Client, options *client.Li
...
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					return err
+				} else {
+					continue
+				}
+			}
+			if foundDADIPlugin && foundNvidiaPlugin {
+				return nil
+			}
```
Comment (Fei-Guo) [inline pkg/node/node.go]: You may move sleep in the end. The thought is that if indeed we need to sleep, we can update status indicating that we are waiting for addon to be ready.

File: pkg/node/node.go (hunk 1 above)
Comment (helayoty) [reply pkg/node/node.go]: Done.

File: pkg/node/node.go (hunk 2 above)
Comment (helayoty) [reply pkg/node/node.go]: I had to move the func to the controller to be able to update the status. Done.

File: api/v1alpha1/workspace_condition_types.go [hunk 3]
```diff
@@ -13,8 +13,11 @@ const (
 	// WorkspaceConditionTypeResourceProvisioned is the state when Resources have been created.
 	WorkspaceConditionTypeResourceProvisioned = ConditionType("ResourceProvisioned")
 
-	// WorkspaceConditionTypeConfigured is the state when Resources have been Configured.
-	WorkspaceConditionTypeConfigured = ConditionType("ResourceConfigured")
+	// WorkspaceConditionTypeMachineStatus is the state when checking machine status.
+	WorkspaceConditionTypeMachineStatus = ConditionType("MachineStatus")
+
+	// WorkspaceConditionTypeInstallNodePlugins is the state when installing node plugins.
+	WorkspaceConditionTypeInstallNodePlugins = ConditionType("InstallNodePlugins")
```
Comment (Fei-Guo) [inline api/v1alpha1/workspace_condition_types.go]: Let us be consistent for the status:
MachineStatus -> MachineIsReady
InstallNodePlugins -> NodePluginsInstalled

Comment (helayoty) [reply api/v1alpha1/workspace_condition_types.go]: Addressed.

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**